### ISSUE TITLE
Building with grunt fails on task ts

### ISSUE THREAD

**Issue by stevelacy** [NativeScript/nativescript-cli#250]
Attempting to run `grunt` produces 508 type warnings:

```

>> 508 non-emit-preventing type warnings  
>> Error: tsc return code: 2
Warning: Task "ts:devlib" failed. Use --force to continue.

Aborted due to warnings.

```
An example of the error:
```
nativescript-cli/lib/services/user-settings-service.ts(14,1): error TS2304: Cannot find name '$injector'.

```

grunt version: 
grunt-cli v0.1.13
grunt v0.4.2

Node: v0.12.0
npm: 2.6.0

I am attempting to build the project to update the dependencies which will fix issues on io.js and node.

**Comment by rosen-vladimirov** [comment]
Hi @stevelacy, I can suggest you to check if you have initialized the submodule. You should execute the following commands:
git clone git@github.com:NativeScript/nativescript-cli.git
cd nativescript-cli
git submodule init
git submodule update
git checkout release
cd lib/common
git checkout release
cd -
grunt
npm install

Basically this should work for you. Could you give it a try?

**Comment by stevelacy** [comment]
Those commands did function though they did seem rather convoluted.

Thanks for clarifying, I probably will open a PR to fix issues on v0.12 which I encountered.

**Comment by rosen-vladimirov** [comment]
@stevelacy I'm glad the commands helped. Please note that you'll have to use release branch where we've included our node 0.12 fixes. We are ready to help in case you find any problems :)
